src/Streamly/FileSystem/Event.hs

```diff
@@ -0,0 +1,239 @@
+utf8ToString = runIdentity . Stream.toList . U.decodeUtf8' . Array.toStream
+#endif
+
+--toUtf8 :: MonadIO m => String -> m (Array Word8)
+--toUtf8 = Array.fromStream . Unicode.encodeUtf8' . Stream.fromList
+
+-------------------------------------------------------------------------------
+-- Main
+-------------------------------------------------------------------------------
+type Toggle = Event.Toggle
+type Config = Event.Config
+type Event = Event.Event
+
+setAccessed :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setAccessed = Event.setModifiedLastWrite
+#elif defined(CABAL_OS_LINUX)
+setAccessed = Event.setAccessed
+#elif defined(CABAL_OS_DARWIN)
+setAccessed = undefined
+#endif
+
+setMetadataChanged :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setMetadataChanged = Event.setModifiedAttribute
+#elif defined(CABAL_OS_LINUX)
+setMetadataChanged = Event.setMetadataChanged
+#elif defined(CABAL_OS_DARWIN)
+setMetadataChanged = undefined
+#endif
+
+setModifiedFileName :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedFileName = Event.setModifiedFileName
+#elif defined(CABAL_OS_LINUX)
+setModifiedFileName s cfg = (Event.setMovedFrom s . Event.setMovedTo s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedFileName = undefined
+#endif
+
+setModifiedDirName :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedDirName = Event.setModifiedDirName
+#elif defined(CABAL_OS_LINUX)
+setModifiedDirName s cfg = (Event.setMovedFrom s . Event.setMovedTo s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedDirName = Event.setRootChanged
+#endif
+
+setModifiedSize :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedSize = Event.setModifiedSize
+#elif defined(CABAL_OS_LINUX)
+setModifiedSize s cfg = (Event.setMetadataChanged s . Event.setModified s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedSize = undefined
+#endif
+
+setModifiedLastWrite :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedLastWrite = Event.setModifiedLastWrite
+#elif defined(CABAL_OS_LINUX)
+setModifiedLastWrite s cfg =
+    (Event.setMetadataChanged s . Event.setModified s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedLastWrite = undefined
+#endif
+
+setModifiedSecurity :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedSecurity = Event.setModifiedSecurity
+#elif defined(CABAL_OS_LINUX)
+setModifiedSecurity s cfg =
+    (Event.setMetadataChanged s . Event.setModified s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedSecurity = undefined
+#endif
+
+
+defaultConfig :: Config
+defaultConfig = Event.defaultConfig
+
+setAllEvents :: Toggle -> Config -> Config
+#if defined(CABAL_OS_DARWIN)
+setAllEvents = defaultConfig
+#else
+setAllEvents = Event.setAllEvents
+#endif
+
+getAbsPath :: Event -> String
+#if defined(CABAL_OS_DARWIN)
+getAbsPath = utf8ToString Event.getAbsPath
+#else
+getAbsPath ev = getRootPath ev </> getRelativePath ev
```

Comment:
We should implement it in the platform specific module itself. Also see an issue about this that I raised earlier.